service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -77,6 +79,9 @@ public RunSetsApiController(
 
   @Override
   public ResponseEntity<RunSetStateResponse> postRunSet(RunSetRequest request) {
+    // request metrics (before validation)
+    captureRequestMetrics(request);
```

Comment:
what do we think about this being (1) it's own function, and (2) before any validation happens?

Reply:
I think it makes sense to keep it as a separate function. For (2) I guess the benefit of capturing here is that we will be able to see how many requests are there with number of inputs and/or outputs higher than our max allowed. So it does make sense to have it here.